extension/app/ts/background/background.ts

```diff
@@ -444,7 +444,16 @@ export function postMessageIfStillConnected(port: browser.runtime.Port, message:
 	const tabId = port.sender?.tab?.id
 	if (tabId === undefined) return false
 	if (!window.interceptor.websiteTabConnection.has(tabId)) return false
-	port.postMessage(message)
+	try {
+		port.postMessage(message)
+	} catch (error) {
+		if (error instanceof Error) {
+			if (error.message?.includes('Attempting to use a disconnected port object')) {
+				return
+			}
+		}
+		throw error
+	}
```

Comment:
I'm pretty strongly against using errors for control flow.  Is there any other way to accomplish this without relying on catching a specific error?  Is there some way we can check the port before posting to it?

For example, can we do something like this:
```ts
port.onDisconnect(() => port = undefined)
```
and then when we use the port do `port?.postMessage(message)`?

Reply:
this error happens in race conditions only so I don't think there's a way

Reply:
Hmm, race conditions shouldn't be possible in JavaScript without async code.  It is *possible* that the `onDisconnect` callback isn't called until the next frame after the disconnect, but that would surprise me a bit.  Usually in JS libraries the callbacks are called inline.